pkg/image/reference/reference.go

```diff
@@ -217,6 +213,15 @@ func (r DockerImageReference) Exact() string {
 	if len(r.Namespace) != 0 {
 		s += r.Namespace + "/"
```

Comment:
If name is not set, appending this slash is also not correct.

however, to Oleg’s point there should never be a scenario where name is empty but namespace is, because in that case name should be set to namespace.  And likewise we can’t trust only registry because some registry values can be valid names (localhost:8080 is a valid name + tag)